include/oxen/quic/utils.hpp

```diff
@@ -202,6 +202,12 @@ namespace oxen::quic
 
     std::string str_tolower(std::string s);
 
+    template <std::integral T>
+    constexpr bool increment_will_overflow(T val)
+    {
+        return std::numeric_limits<T>::max() - val <= 1;
```

Comment:
```suggestion
        return std::numeric_limits<T>::max() == val;
```

Reply:
This function is explicitly meant to check if it will overflow after the increment. Re-naming it to fit the purpose was something you brought up earlier, and this change muddies the water in that